stablehlo/dialect/Serialization.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_ASSEMBLYFORMAT_H
+#define STABLEHLO_DIALECT_ASSEMBLYFORMAT_H
+
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir {
+namespace stablehlo {
+
+// Write a StableHLO program to a portable artifact
+LogicalResult serializePortableArtifact(ModuleOp module,
+                                        std::string const& targetVersion,
+                                        raw_ostream& os);
+
+// Read StableHLO portable artifact
+mlir::OwningOpRef<mlir::ModuleOp> deserializePortableArtifact(
+    llvm::StringRef sourceStr, MLIRContext* context);
+
+}  // namespace stablehlo
+}  // namespace mlir
+
+#endif  // STABLEHLO_DIALECT_ASSEMBLYFORMAT_H
```

Comment:
Let's also update documentation?
  1) compatibility.md to point to this file instead of the VHLO cookbook (so that VHLO stays an implementation detail from the point of view of compatibility.md).
  2) vhlo.md to use `stablehlo-interpreter` instead of `stablehlo-opt`.

Reply:
Lets talk more about this. I updated (2), but given the suggested update to (2) I think that updating (1) would
result in providing less useful information, or redundant information.

Reply:
Sounds good.